### PR TITLE
Try to guess FPU stack delta of call instructions.

### PR DESCRIPTION
**Pull request by ptomin**
FPU stack delta of hell nodes is unknown. But we can guess it based on
FPU stack variable uses after call.
If we have
```
    call eax_1()
    defs: Top_3
    Extract return from FPU stack
    edx_4 = ST_2[Top_3]
    Top_5 = Top_3 + 1
```
then we can suppose that FPU stack delta is `-1`. Of course,
`ST_2[Top_3]` could be defined before indirect call, but it's unlikely

### CODE REVIEW

File: src/Decompiler/Analysis/FpuStackReturnGuesser.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        }
+
+        public void Rewrite()
+        {
+            var fpuStack = ssa.Procedure.Architecture.FpuStackRegister;
+            if (fpuStack == null)
+                return;
+            var fpuStackIds = ssa.Identifiers
+                .Where(sid => sid.Identifier.Storage == fpuStack)
+                .Where(sid => sid.DefStatement != null)
```
Comment (uxmal) [inline src/Decompiler/Analysis/FpuStackReturnGuesser.cs]: Consider fusing the two `Where` calls to a single expression:
`.Where(sid => sid.DefStatement != null && sid.Identifier.Storage == fpuStack)`

File: src/Decompiler/Analysis/FpuStackReturnGuesser.cs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    /// <example>
+    /// If we have
+    /// <code>
+    ///     call eax_1()
+    ///         defs: Top_3
+    ///     // Extract return from FPU stack
+    ///     edx_4 = ST_2[Top_3]
+    ///     Top_5 = Top_3 + 1
+    /// </code>
+    /// then we can suppose that FPU stack delta is -1. Of course,
```
Comment (uxmal) [inline src/Decompiler/Analysis/FpuStackReturnGuesser.cs]: Minor quibble: use `assume` rather than `suppose`.

Comment (ptomin) [reply src/Decompiler/Analysis/FpuStackReturnGuesser.cs]: Done. See https://github.com/uxmal/reko/compare/a1d1fe1733e0c22ecb2c53b956913a801a128fb2..fb1d68656439a4435c9d99b11028745c6b00220a

File: src/Decompiler/Analysis/FpuStackReturnGuesser.cs (hunk 1 above)
Comment (ptomin) [reply src/Decompiler/Analysis/FpuStackReturnGuesser.cs]: Done. See https://github.com/uxmal/reko/compare/a1d1fe1733e0c22ecb2c53b956913a801a128fb2..fb1d68656439a4435c9d99b11028745c6b00220a

**Pull request merged**